javascript/code-challenges/challenges-11.test.js

```diff
@@ -0,0 +1,224 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named transformToLis that, given an object, returns an array of the key value pairs as html list items.
+
+For example:
+{
+  name: 'bob',
+  age: 32
+}
+
+Becomes:
+[
+<li>name: bob</li>,
+<li>age: 32</li>
+]
+------------------------------------------------------------------------------------------------ */
+
+function transformToLis(obj){
+  let keys = Object.keys(obj);
+  return keys.map(key => {
+    return `<li>${key}: ${obj[key]}</li>`;
+  });
+}
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named count that, given an integer and an array of arrays, uses either filter, map, or reduce to count the amount of times the integer is present in the array of arrays.
+
+Note: You might need to use the same method more than once.
+
+For example, count(5, [[1, 3, 5, 7, 9], [5, 5, 5], [1, 2, 3]]) returns 4.
+------------------------------------------------------------------------------------------------ */
+
+const count = (target, input) => {
+  let num = 0;
+  input.map(arr => {
+    arr.map(value => {
+      if(value === target){
+        num++;
+      }
+    });
+  });
+  return num;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function that, given an array of integer arrays as input, calculates the total sum of all the elements in the array.
+
+You may want to use filter, map, or reduce for this problem, but are not required to. You may need to use the same method more than once.
+
+For example, [[1, 2, 3, 4, 5], [6, 7, 2, 4, 5, 7], [9, 2, 3, 6,]] returns 66.
+------------------------------------------------------------------------------------------------ */
+
+const totalSum = (input) => {
+  let sum = 0;
+  input.map( arr => {
+    arr.map(value => sum += value);
+  });
+  return sum;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named divisibleByFiveTwoToThePower that accepts an array of arrays as input.
+
+This function should first remove any elements that are not numbers or are not divisible by five.
+
+This function should then raise 2 to the power of the resulting numbers, returning an array of arrays.
+
+For example, [ [0,2,5,4], [2,4,10], [] ] should return [ [1, 32], [1024], [] ].
+------------------------------------------------------------------------------------------------ */
+
+const divisibleByFiveTwoToThePower = (input) => {
+  // Solution code here...
+};
+
```

Comment:
Good work on challenges!

You mentioned you had a question about this one:

- our input is array of arrays
- 1st we have to remove elements that are not nums/not divisible by 5 (we will start there)
- we can use map() since we need to return a value (an array of nums). We will map over the current array (which holds nums) first `return input.map( currentArr => {`
- Inside our map(), we can make use of reduce, passing in `[]` as our initial value 
   - if the current thing we are on at each iteration ( the`curr` in reduce,  which is a number), is divisible by 5 (we can use `%` modulo for this) and is not a number (see [typeof operator in JS](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof) , then we can move on to raising 2 the power of the resulting nums.